hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/INodeMap.java

```diff
@@ -36,22 +38,22 @@
  * and INode.  
  */
 public class INodeMap {
-  static final int NAMESPACE_KEY_DEPTH = 2;
-  static final int NUM_RANGES_STATIC = 256;  // power of 2
+  private static int namespaceKeyDepth;
+  private static long numRangesStatic;
```

Comment:
This is definitely one way to do it but is there a way we can make numSpaceKeyDepth/numRangesStatic a non-static variable? 

numspaceKeyDepth -> numSpaceKeyDepth 
numRangesStatic -> numRanges

Reply:
Thanks @cxorm  for the comment.
I will update it later.

Reply:
Thanks @xinglin for the comment.
I will update it later.

Reply:
Also, if we want to support namespaceKeyDepth other than 2, we probably need to modify the range Keys we insert when we create new partitions. Instead of inserting range key such as [0, 16385], [1, 16385], [2, 16385], I think we might need to insert range keys as [0,0, 16385], [1,0,16385], [2,0,16385] ... for depth of 3 and [0,0,0,16385], [1,0,0,16385], [2,0,0,16385]... for depth of 4.

```
    for (int p = 0; p < numRangesStatic; p++) {
      INodeDirectory key = new INodeDirectory(INodeId.ROOT_INODE_ID,
          "range key".getBytes(StandardCharsets.UTF_8), perm, 0);
      key.setParent(new INodeDirectory((long)p, null, perm, 0));
```

Reply:
Thanks @xinglin for the comment.
I agree with you and I will update it later.